src/btree/col_srch.c

```diff
@@ -254,6 +254,14 @@ __wt_col_search(
         if ((cip = __col_var_search(current, recno, NULL)) == NULL) {
             cbt->recno = __col_var_last_recno(current);
             cbt->slot = page->entries == 0 ? 0 : page->entries - 1;
+            if (cbt->recno != WT_RECNO_OOB) {
+                ins_head = WT_COL_UPDATE_SLOT(page, cbt->slot);
+                ins = WT_SKIP_LAST(ins_head);
+                if (ins != NULL && cbt->recno == WT_INSERT_RECNO(ins)) {
+                    cbt->ins_head = ins_head;
+                    cbt->ins = ins;
+                }
+            }
```

Comment:
We should also improve test checkpoint to compare the return values to detect not only the not found case but also the mismatch case.

Reply:
I had a go at doing this, but ran into an unrelated issue with the value occasionally being different. It seems to happen the same amount both on this branch and develop. It seems to be a separate issue to this, so if possible I'd like to raise another ticket for it instead of bundling it into this PR?

Reply:
Made WT-9571 for this.